crates/cargo-wdk/src/actions/build/build_task.rs

```diff
@@ -30,47 +29,42 @@ pub struct BuildTask<'a> {
 }
 
 impl<'a> BuildTask<'a> {
-    /// Creates a new instance of `BuildTask`
+    /// Creates a new instance of `BuildTask`.
+    ///
     /// # Arguments
     /// * `package_name` - The name of the package to build
     /// * `working_dir` - The working directory for the build
     /// * `profile` - An optional profile for the build
     /// * `target_arch` - The target architecture for the build
     /// * `verbosity_level` - The verbosity level for logging
     /// * `command_exec` - The command execution provider
-    /// * `fs` - The file system provider
+    ///
     /// # Returns
-    /// * `Result<Self, BuildTaskError>` - A result containing the new instance
-    ///   of `BuildTask` or an error
-    /// # Errors
-    /// * `BuildTaskError::CanonicalizeManifestPath` - If there is an IO error
-    ///   while canonicalizing the working dir
-    /// * `BuildTaskError::EmptyManifestPath` - If the manifest path is empty
+    /// * `Self` - A new instance of `BuildTask`.
+    ///
+    /// # Panics
+    /// * If `working_dir` is not absolute
     pub fn new(
         package_name: &'a str,
         working_dir: &'a Path,
         profile: Option<&'a Profile>,
         target_arch: TargetArch,
         verbosity_level: clap_verbosity_flag::Verbosity,
         command_exec: &'a CommandExec,
-        fs: &'a Fs,
-    ) -> Result<Self, BuildTaskError> {
-        let manifest_path = fs.canonicalize_path(&working_dir.join("Cargo.toml"))?;
-        let manifest_path = match manifest_path.strip_extended_length_path_prefix() {
-            Ok(path) => path,
-            Err(StripExtendedPathPrefixError::NoExtendedPathPrefix) => manifest_path,
-            Err(StripExtendedPathPrefixError::EmptyPath) => {
-                return Err(BuildTaskError::EmptyManifestPath);
-            }
-        };
-        Ok(Self {
+    ) -> Self {
+        assert!(
+            working_dir.is_absolute(),
```

Comment:
We already have error propagation set up from `build_task` to `build_action` to `cli` to `main` module with ability to print nested errors.

This particular line is a fallible call and we should have simply leveraged the existing error propagation by introducing an error type for this case instead of panicking deep inside.